auth/authzserver/claims_verifier.go

```diff
@@ -0,0 +1,64 @@
+package authzserver
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/ory/x/jwtx"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/flyteorg/flyteadmin/auth"
+	"github.com/flyteorg/flyteadmin/auth/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/service"
+)
+
+func verifyClaims(expectedAudience sets.String, claimsRaw map[string]interface{}) (interfaces.IdentityContext, error) {
+	claims := jwtx.ParseMapStringInterfaceClaims(claimsRaw)
+
+	foundAudIndex := -1
+	for audIndex, aud := range claims.Audience {
+		if expectedAudience.Has(aud) {
+			foundAudIndex = audIndex
+			break
+		}
+	}
+
+	if foundAudIndex < 0 {
+		return nil, fmt.Errorf("invalid audience [%v]", claims)
+	}
+
+	userInfo := &service.UserInfoResponse{}
+	if userInfoClaim, found := claimsRaw[UserIDClaim]; found && userInfoClaim != nil {
+		userInfoRaw := userInfoClaim.(map[string]interface{})
+		raw, err := json.Marshal(userInfoRaw)
+		if err != nil {
+			return nil, err
+		}
+
+		if err = json.Unmarshal(raw, userInfo); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal user info claim into UserInfo type. Error: %w", err)
+		}
+	}
+
+	clientID := ""
+	if clientIDClaim, found := claimsRaw[ClientIDClaim]; found {
+		clientID = clientIDClaim.(string)
+	}
+
+	scopes := sets.NewString()
+	if scopesClaim, found := claimsRaw[ScopeClaim]; found {
+		if scopesSlice, ok := scopesClaim.([]interface{}); ok {
+			scopes = sets.NewString(interfaceSliceToStringSlice(scopesSlice)...)
+		} else {
```

Comment:
should you also check the type here if it's a string

Reply:
According to the RFC , this should be slice of strings https://datatracker.ietf.org/doc/html/draft-ietf-oauth-token-exchange-06#section-4.2
But in our case auth server was sending as single value and it can be interpreted as string. So i don't think we need another type check there

Reply:
should we still check the type though before assuming that it'll always be a string if not `[]interface{}`?

Reply:
Done